server/src/com/thoughtworks/go/config/GoConfigWriteLock.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2016 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.config;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class GoConfigWriteLock {
```

Comment:
Even though this object will be a singleton because Spring will make it so, I'd make the writeLock object an explicit static object, just to make sure that there is no way for this lock to be created twice. I say this because this object is being across classes for synchronization.

Or, you can always do a:

```
synchronized (GoConfigWriteLock.class)
```

and be sure that it will lock on the (always singleton - in a classloader) class itself. No need for the method at all.